controller/proxy-injector/webhook.go

```diff
@@ -53,8 +62,23 @@ func Inject(api *k8s.API,
 		Allowed: true,
 	}
 
-	if !report.Injectable() {
-		log.Infof("skipped %s", report.ResName())
+	var parent *runtime.Object
+	if ownerRef := resourceConfig.GetOwnerRef(); ownerRef != nil {
+		objs, err := api.GetObjects(request.Namespace, ownerRef.Kind, ownerRef.Name)
+		if err != nil {
+			return nil, err
+		}
+		if len(objs) == 0 {
+			return nil, fmt.Errorf("couldn't retrieve parent object %s-%s-%s", request.Namespace, ownerRef.Kind, ownerRef.Name)
+		}
+		parent = &objs[0]
+	}
+
+	if injectable, reason := report.Injectable(); !injectable {
+		if parent != nil {
+			recorder.Eventf(*parent, v1.EventTypeNormal, eventTypeSkipped, "Linkerd sidecar proxy injection skipped: %s", reason)
```

Comment:
if there's no parent, should we record the event with the object itself?

Reply:
At this point in the persistence workflow the pod hasn't been persisted. If one tries to associate the event with the pod, the api will error out saying it can't find the object.